### PR TITLE
Migrate `manager/clusters/join-cluster`

### PR DESCRIPTION
**Pull request by Etheryte**
## What does this PR change?

Migrate all components in `manager/clusters/join-cluster` to Typescript.  
This is the migration I did in the review meeting demo.  

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Part of https://github.com/SUSE/spacewalk/issues/13741

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

**Review by ncounter — APPROVED** [review]

**Review by cbbayburt — APPROVED** [review]
Looks good :+1: 

Just a couple of nitpicks as usual :stuck_out_tongue:

File: web/html/src/manager/clusters/join-cluster/join-cluster.tsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+import { MessageType, ServerMessageType } from "components/messages";
+
+type Props = {
+  cluster: ClusterType;
+  flashMessage?: ServerMessageType;
+  setMessages: (messages: MessageType[]) => void;
+};
+
+const JoinCluster = (props: Props) => {
+  const [joinConfig, setJoinConfig] = useState<FormulaValuesType | null | undefined>(null);
```
[the remark this answers is not in the document]
Comment (mbologna) [reply web/html/src/manager/clusters/join-cluster/join-cluster.tsx:24]: :+1:

File: web/html/src/manager/clusters/join-cluster/join-cluster.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+  const [nodesToJoin, setNodesToJoin] = useState<Array<ServerType>>([]);
+
+  const { fetchNodesToJoin, scheduleJoinNode } = useClustersApi();
+
+  const scheduleJoin = (earliest: Date, actionChain: string | null | undefined): Promise<any> => {
```
[the remark this answers is not in the document]
Comment (mbologna) [reply web/html/src/manager/clusters/join-cluster/join-cluster.tsx:29]: :+1:

**Review by mbologna — APPROVED** [review]
Apart from the two comments that need resolution before merging, it is looking good. Thanks for the effort!

**Comment by Etheryte** [comment]
Thanks a lot for helping constrain the types down further, I've merged the changes and cleaned up the related connections.

**Pull request merged**